### PR TITLE
Change Torch wheel install variants for CI

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** The existing CI infra uses Torch wheels pulled directly from the PyTorch hosted builds. This has defaulted to pulling in the ROCm build of the wheels, which are 2GB to download, and 12+GB when installed and unpacked. As Github action only provide 14GB guaranteed storage for runners, this immediately eats into the available storage and often causes out of space failures.

**Description of the Change:** This changes the Torch wheel installs to the default ones from PyPI. The default wheels still pull in a large number of external deps (several GB, including Nvidia runtime libs), but as of now allows us to remain below the 14GB threshold.

**Benefits:** Prevent out of space errors in the CI.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Very good call! Thank you for that.

File: pennylane/workflow/__init__.py [hunk 1]
```diff
@@ -29,19 +29,6 @@
...
-.. autosummary::
-    :toctree: api
-
-    ~workflow.interfaces.autograd
-    ~workflow.interfaces.jax
-    ~workflow.interfaces.jax_jit
-    ~workflow.interfaces.tensorflow
-    ~workflow.interfaces.tensorflow_autograph
-    ~workflow.interfaces.torch
-
```
Comment (mudit2812) [inline pennylane/workflow/__init__.py:44]: Why was this removed?

**Comment by mudit2812** [comment]
I suggest adding the "urgent" label.

**Review by PietropaoloFrisoni — APPROVED** [review]

File: pennylane/workflow/__init__.py (hunk 1 above)
Comment (mlxd) [reply pennylane/workflow/__init__.py:44]: @albi3ro

Comment (albi3ro) [reply pennylane/workflow/__init__.py:44]: We were rendering the documentation for a torch object that has deprecation warnings in it's docstrings. Or something.

Basically, these things are really internal and don't really need to render in the docs, and trying to render them caused issues.
